app/models/concerns/submit_result.rb

```diff
@@ -0,0 +1,56 @@
+class SubmitResult
+  module SubmitResultType
+    SUCCESSFULL = "successfull"
+    HAS_SYNTAX_ERRORS = "has_syntax_errors"
+    PENDING_REFACTORING = "pending_refactoring"
+  end
+
+  attr_reader :syntax_errors
+  attr_reader :type
+
+  def initialize answer, keywords_hash
+    @syntax_errors = SubmitResult.findSyntaxErrors(answer)
+    @type = determineSubmitResult(answer, keywords_hash)
```

Comment:
Usually the constructor is supposed to just build up the object in the initial state. So this space is usually reserved for setting instance variables from what was passed in to the `new` method. Is there a way you can separate out these method calls in a different place?